### ISSUE TITLE
OpenShift.io users inactive for a long period of time adjusted to consume a minimal amount of OSO Starter resources

### ISSUE THREAD

**Issue by qodfathr** [openshiftio/openshift.io#2029]
OpenShift.io users who have been inactive for an extended period of time -- currently defined as 8 weeks but open for review and discussion -- should have their OpenShift Online Starter resource consumption reduced to the absolute minimum. However, should any of these users return to OpenShift.io, they must be put back into a state essentially the same as when they last logged in.

In essence, all of the user's OpenShift Online Starter resources should be dehydrated and removed from the operational cluster. This takes their applications offline, removes their pipelines, etc. They can no longer log into the OpenShift Online console.

The user's OpenShift.io resources remain intact -- their account, their Spaces, their Work Items, etc. However, anything displayed in OpenShift.io which actually lives in OSO will not be available. This includes, but is not limited to, their Pipelines, their Che Workspaces, their Environments, etc.

The URLs for the user's account and Spaces will continue to exist and operate. However, if someone other than the account owner visits a Screen which requires resources from OpenShift Online, those resources are *not* brought back online. For example, if user Alice has been inactive so long that her OpenShift Online Starter environment has been dehydrated, then if user Bob visits the Pipeline screen of one of Alice's Spaces., the list of pipelines will be empty and Alice's pipelines are not automatically recreated.

If the dehydrated user logs back into OpenShift.io, their OpenShift Online Starter environment will be automatically hydrated. The hydration process can be asynchronous and may take a substantial amount of time to complete. Any OpenShift.io screen impacted by a currently running hydration process should include some sort of notification message. For example, if user Alice was inaction for so long that her OSO Starter environment was dehydrated, but then Alice returns, when Alice first logs in she should be alerted that her Environments need to be recreated and this may take an extended amount of time. If Alice navigates to the Pipeline screen and the hydration process is not yet complete, she should see a message informing her that her environment is still being recreated.

## Parent Scenario
#2028 OpenShift.io scales to meet user demand

**Comment by maxandersen** [comment]
> the list of pipelines will be empty and Alice's pipelines are not automatically recreated.

do you really mean that ? wouldn't the more natural thing be that the pipelines are listed there but marked as "Inactive" or similar. Related scenario are when a user that don't have RBAC approved access to the cluster - he/she won't be able to see anything either as they simply don't have the access. In that case it would also  be relevant to have some status/list of pipelines that can be viewed without dev-user level access to the openshift online cluster.

**Comment by ldimaggi** [comment]
What would the problem be with the user recreating the pipeline? Couldn't he/she just "create pipeline" from OSIO?

**Comment by maxandersen** [comment]
> What would the problem be with the user recreating the pipeline? Couldn't he/she just "create pipeline" from OSIO?

not following your question - sorry. can you provide a bit more context ?

**Comment by qodfathr** [comment]
@maxandersen yes, an indication of "Inactive" makes more sense then empty. I'll revise the wording. Thanks.
